test/services/api/api.service.unit.ts

```diff
@@ -0,0 +1,39 @@
+import { createOAuthApiClient } from "../../../src/services/api/api.service";
+import { getEmptySessionRequest, getSessionRequest } from "../../mocks/session.mock";
+import { Session } from "@companieshouse/node-session-handler";
+
+const ERROR_MESSSAGE = "Error getting session keys for creating api client";
+
+describe ("Test node session handler authorization for private sdk", () => {
+  it ("should obtain private node sdk oauth client", () => {
+    const client = createOAuthApiClient(
+      getSessionRequest({ access_token: "token" }));
+    expect(client.confirmationStatementService).not.toBeNull();
+  });
+
+  it("should throw error when no data is present", () => {
+    try {
+      createOAuthApiClient(getEmptySessionRequest());
+    } catch (error) {
+      expect(error.message).toBe(ERROR_MESSSAGE);
+    }
+  });
+
+  it("should throw error when no sign in info is present", () => {
+    try {
+      const session: Session = getEmptySessionRequest();
+      session.data = {};
+      createOAuthApiClient(session);
```

Comment:
just needs a fail() after the call to the function (in each error test) so that the test won't pass if no error is thrown